src/pipert2/core/base/method.py

```diff
@@ -0,0 +1,77 @@
+from typing import Optional
+from src.pipert2.utils.consts.event_format import EVENT_SEPARATOR, EVENT_INDEX, FLOW_INDEX, EVENT_CONTAIN_ROUTINE_LENGTH, \
+    ROUTINE_INDEX
+
+
+class Method:
+    """The method class uses for describing event command and his parameters.
+    Using methods we notify on events through the pipeline components.
+    """
+
+    def __init__(self, name: str, params: dict = {}):
+        """
+        Args:
+            name (str): The method name in format - { event-flow_name-routine_name }.
+            params (dict): Dictionary of method parameters, default is empty dictionary.
+        """
+
+        self.name = name
+        self.params = params
```

Comment:
IMO you should add a new data member for the event locations.
For example save in one dm the name and on the other one save the extensions of it.

It will make it more simple, and you won't have to create a new Method with only the name.

(ofc you might need to change the name of the `name` parameter to something else)

What do you say ?